app/controllers/application_controller.rb

```diff
@@ -1,2 +1,24 @@
 class ApplicationController < ActionController::API
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+
+  before_action :authenticate_token
+
+  private
+
+    def authenticate_token
+      unless authentication_status == :ok
+        render json: URIService::JSON.errors(authentication_status.to_s.titlecase),
+               status: authentication_status
+      end
+    end
+
+    def authentication_status
+      status = :unauthorized
+      authenticate_with_http_token do |token, options|
+        Rails.application.config_for(:uri_service).fetch('api_keys', []).tap do |valid_api_key|
```

Comment:
Can we store `Rails.application.config_for(:uri_service)` in a constant (via some initializer) so that we don't need to re-read in the yml file each time?